platform/admin/core/admin-core-alertservice-data/src/main/java/org/codice/ddf/admin/core/alert/service/data/internal/Alert.java

```diff
@@ -0,0 +1,168 @@
+        this(level, title, details);
+        setKey(key);
+    }
+
+    /**
+     * Constructs an {@link Alert} without a key
+     *
+     * @param level   non-null {@value #LEVEL}
+     * @param title   non-null, not empty {@value #TITLE}
+     * @param details non-null (may be empty) list of String {@value #DETAILS}
+     */
+    public Alert(@Nonnull Level level, @Nonnull @NotEmpty String title,
+            @Nonnull List<String> details) {
+        setLevel(level);
+        setTitle(title);
+        setDetails(details);
+    }
+
+    /**
+     * Sets the {@value #KEY}
+     *
+     * @param key non-null, non-empty {@value #KEY}
+     */
+    public void setKey(@Nonnull @NotEmpty String key) {
+        notEmpty(key, "key may not be empty");
+        this.put(KEY, key);
+    }
+
+    /**
+     * This may return null if the {@link #Alert(String, Level, String, List)} method was used to
+     * construct the {@link Alert}. If the returned key is not null, it should not be blank.
+     *
+     * @return the nullable {@value #KEY} of the {@link Alert}
+     */
+    @Nullable
+    public String getKey() {
+        return (String) this.get(KEY);
+    }
+
+    /**
+     * Sets the {@value #LEVEL}
+     *
+     * @param level non-null {@value #LEVEL}
+     */
+    private void setLevel(@Nonnull Level level) {
+        notNull(level, "level may not be null");
+        this.put(LEVEL, level.toString());
+    }
+
+    /**
+     * @return the non-null {@value #LEVEL} of the {@link Alert}
+     */
+    @Nonnull
+    public String getLevel() {
+        return (String) this.get(LEVEL);
+    }
+
+    /**
+     * Sets the {@value #TITLE}
+     *
+     * @param title non-null, not empty {@value #TITLE}
+     */
+    private void setTitle(@Nonnull @NotEmpty String title) {
+        notEmpty(title, "title may not be empty");
+        this.put(TITLE, title);
+    }
+
+    /**
+     * @return the non-null, not empty {@value #TITLE} of the {@link Alert}
+     */
+    @Nonnull
+    @NotEmpty
+    public String getTitle() {
+        return (String) this.get(TITLE);
+    }
+
+    /**
+     * Sets the {@value #DETAILS} of the {@link Alert}
+     *
+     * @param details non-null (may be empty) list of String {@value #DETAILS}
+     */
+    private void setDetails(@Nonnull List<String> details) {
+        notNull(details, "details may not be null");
+        this.put(DETAILS, details);
+    }
+
+    /**
+     * @return non-null (may be empty) list of String {@value #DETAILS} of the {@link Alert}
+     */
+    @Nonnull
+    public List<String> getDetails() {
+        return (List<String>) this.get(DETAILS);
+    }
+
+    public enum Level {
+        INFO {
+            public String toString() {
+                return "info";
+            }
+        }, WARN {
+            public String toString() {
+                return "warning";
+            }
+        }, ERROR {
+            public String toString() {
+                return "danger";
+            }
```

Comment:
I extracted a method and added a comment in AlertServiceBean to explain why the enums are mapped to different strings (e.g. "ERROR" -> "danger").
```java
    /**
     * Maps the enums to existing Admin Console styles
     */
    private static String getLevelString(Alert alert) {
    ...
```